daprdocs/content/en/dotnet-sdk-docs/dotnet-actors/dotnet-actors-serialization.md

```diff
@@ -0,0 +1,315 @@
+
+```xml
+<Doodad>
+  <Id>a06ced64-4f42-48ad-84dd-46ae6a7e333d</Id>
+  <Name>DoodadName</Name>
+  <Count>5</Count>
+</Doodad>
+```
+
+So let's tweak it - let's add our own constructor and only use init-only setters on the members. This will fail to serialize and deserialize not because of the use of the init-only setters, but because there's no parameterless constructors.
+
+```csharp
+// WILL NOT SERIALIZE PROPERLY!
+public class Doodad
+{
+    public Doodad(string name, int count)
+    {
+        Id = Guid.NewGuid();
+        Name = name;
+        Count = count;
+    }
+
+    public Guid Id { get; set; }
+    public string Name { get; init; }
+    public int Count { get; init; }
+}
+```
+
+If we add a public parameterless constructor to the type, we're good to go and this will work without further annotations. 
+
+```csharp
+public class Doodad
+{
+    public Doodad()
+    {
+    }
+
+    public Doodad(string name, int count)
+    {
+        Id = Guid.NewGuid();
+        Name = name;
+        Count = count;
+    }
+
+    public Guid Id { get; set; }
+    public string Name { get; set; }
+    public int Count { get; set; }
+}
+```
+
+But what if we don't want to add this constructor? Perhaps you don't want your developers to accidentally create an instance of this Doodad using an unintended constructor. That's where the more flexible attributes are useful. If you decorate your type with a [DataContractAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.serialization.datacontractattribute) attribute, you can drop your parameterless constructor and it will work once again.
+
+```csharp
+[DataContract]
+public class Doodad
+{
+    public Doodad(string name, int count)
+    {
+        Id = Guid.NewGuid();
+        Name = name;
+        Count = count;
+    }
+
+    public Guid Id { get; set; }
+    public string Name { get; set; }
+    public int Count { get; set; }
+}
+```
+
+In the above example, we don't need to also use the [DataMemberAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.serialization.datamemberattribute) attributes because again, we're using [built-in primitives](###supported-primitive-types) that the serializer supports. But, we do get more flexibility if we use the attributes. From the DataContractAttribute attribute, we can specify our own XML namespace with the Namespace argument and, via the Name argument, change the name of the type as used when serialized into the XML document. 
+
+It's a recommended practive to append the DataContractAttribute attribute to the type and the DataMemberAttribute attributes to all the members you want to serialize anyway - if they're not necessary and you're not changing the default values, they'll just be ignored, but they give you a mechanism to opt into serializing members that wouldn't otherwise have been included such as those marked as private or that are themselves complex types or collections.
```

Comment:
Spelling: `practice`

Reply:
Spell check is an excellent idea. Fixed a few additional misspellings as well.